pkg/virtual/framework/dynamic/apiserver/apiserver.go

```diff
@@ -0,0 +1,184 @@
+	APISetRetriever  apidefs.APISetRetriever
+}
+
+type completedConfig struct {
+	GenericConfig genericapiserver.CompletedConfig
+	ExtraConfig   *DynamicAPIServerExtraConfig
+}
+
+type CompletedConfig struct {
+	// Embed a private pointer that cannot be instantiated outside of this package.
+	*completedConfig
+}
+
+// Complete fills in any fields not set that are required to have valid data. It's mutating the receiver.
+func (c *DynamicAPIServerConfig) Complete() completedConfig {
+	cfg := completedConfig{
+		c.GenericConfig.Complete(),
+		&c.ExtraConfig,
+	}
+
+	return cfg
+}
+
+// New returns a new instance of DynamicAPIServer from the given config.
+func (c completedConfig) New(virtualWorkspaceName string, delegationTarget genericapiserver.DelegationTarget) (*DynamicAPIServer, error) {
+	genericServer, err := c.GenericConfig.New(virtualWorkspaceName+"-virtual-workspace-apiserver", delegationTarget)
+	if err != nil {
+		return nil, err
+	}
+
+	director := genericServer.Handler.Director
+	genericServer.Handler.Director = http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+		if vwName := r.Context().Value(virtualcontext.VirtualWorkspaceNameKey); vwName != nil {
+			if vwNameString, isString := vwName.(string); isString && vwNameString == virtualWorkspaceName {
+				director.ServeHTTP(rw, r)
+				return
+			}
+		}
+		delegatedHandler := delegationTarget.UnprotectedHandler()
+		if delegatedHandler != nil {
+			delegatedHandler.ServeHTTP(rw, r)
+		} else {
+			http.NotFoundHandler().ServeHTTP(rw, r)
+		}
+	})
+
+	s := &DynamicAPIServer{
+		GenericAPIServer: genericServer,
+		APISetRetriever:  c.ExtraConfig.APISetRetriever,
+	}
+
+	delegateHandler := delegationTarget.UnprotectedHandler()
+	if delegateHandler == nil {
+		delegateHandler = http.NotFoundHandler()
+	}
+
+	versionDiscoveryHandler := &versionDiscoveryHandler{
+		apiSetRetriever: s.APISetRetriever,
+		delegate:        delegateHandler,
+	}
+
+	groupDiscoveryHandler := &groupDiscoveryHandler{
+		apiSetRetriever: s.APISetRetriever,
+		delegate:        delegateHandler,
+	}
+
+	rootDiscoveryHandler := &rootDiscoveryHandler{
+		apiSetRetriever: s.APISetRetriever,
+		delegate:        delegateHandler,
+	}
+
+	crdHandler, err := newResourceHandler(
+		s.APISetRetriever,
+		versionDiscoveryHandler,
+		groupDiscoveryHandler,
+		rootDiscoveryHandler,
+		delegateHandler,
+		c.GenericConfig.AdmissionControl,
+		s.GenericAPIServer.Authorizer,
+		c.GenericConfig.RequestTimeout,
+		time.Duration(c.GenericConfig.MinRequestTimeout)*time.Second,
+		c.GenericConfig.MaxRequestBodyBytes,
+		s.GenericAPIServer.StaticOpenAPISpec,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	s.GenericAPIServer.Handler.GoRestfulContainer.Filter(func(req *restful.Request, resp *restful.Response, chain *restful.FilterChain) {
+		pathParts := splitPath(req.Request.URL.Path)
+		if len(pathParts) > 0 && pathParts[0] == "apis" ||
+			len(pathParts) > 1 && pathParts[0] == "api" {
```

Comment:
Because both `/apis` and `/api/v1`, as well as paths starting with these, should be managed by the `crdHandler`.
But `/api` should still be managed by the `LegacyRootHandler` (as seen just below).